.circleci/config.yml

```diff
@@ -1,175 +1,142 @@
-  testdotty_jdk8:
-    docker:
-      - image: hseeberger/scala-sbt:8u181_2.12.8_1.2.8
-    steps:
-      - checkout
-      - restore_cache:
-          key: sbt-cache
-      - run: sbt ++0.13.0-RC1! coreJVM/compile coreJVM/package coreJVM/packageSrc coreJVM/publishLocal
-      - save_cache:
-          key: sbt-cache
-          paths:
-            - "~/.ivy2/cache"
-            - "~/.sbt"
-            - "~/.m2"
-            - "~/website/node_modules"
-  release:
-    docker:
-      - image: hseeberger/scala-sbt:8u181_2.12.8_1.2.8
-    steps:
-      - checkout
-      - run: git fetch --tags
-      - restore_cache:
-          key: sbt-cache
-      - run: echo -n "${PGP_PUBLIC}" | base64 -d > /tmp/public.asc
-      - run: echo -n "${PGP_SECRET}" | base64 -d > /tmp/secret.asc
-      - run: sbt ++2.11.12! releaseEarly
-      - run: sbt ++2.12.8! releaseEarly
-  documentation:
-    docker:
-      - image: hseeberger/scala-sbt:8u181_2.12.8_1.2.8
-    steps:
+
+scala_211: &scala_211
+  SCALA_VERSION: 2.11.12
+
+scala_212: &scala_212
+  SCALA_VERSION: 2.12.8
+
+scala_dotty: &scala_dotty
+  SCALA_VERSION: 0.13.0-RC1
+
+jdk_8: &jdk_8
+  JDK_VERSION: 8
+
+jdk_11: &jdk_11
+  JDK_VERSION: 11
+
+machine_ubuntu: &machine_ubuntu
+  machine:
+    image: ubuntu-1604:201903-01
+
+install_jdk: &install_jdk
+  - run:
+      name: Install JDK
+      command: |
+        while $(ps aux | grep -i ' apt ' | grep -v grep > /dev/null); do sleep 1; done # Wait for apt to be ready
+        sudo add-apt-repository ppa:openjdk-r/ppa -y
+        sudo apt update
+        sudo apt install openjdk-${JDK_VERSION}-jdk -y
+
+load_cache: &load_cache
+  - restore_cache:
+      key: sbt-cache
+
+save_cache: &save_cache
+  - save_cache:
+      key: sbt-cache
+      paths:
+          - "~/.ivy2/cache"
+          - "~/.sbt"
+          - "~/.m2"
+          - "~/website/node_modules"
+
+install_yarn: &install_yarn
+  - run:
+      name: Install Yarn
+      command: |
+        curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+        echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list
+        sudo apt update && sudo apt install yarn -y
+
+install_nodejs: &install_nodejs
+  - run:
+      name: Install node.js
+      command: |
+        curl -sL https://deb.nodesource.com/setup_10.x | sudo bash -
+        sudo apt install -y nodejs
+
+lint: &lint
+  steps:
+    - checkout
+    - <<: *load_cache
+    - run:
+        name: Lint code
+        command: ./sbt ++${SCALA_VERSION}! check
+    - <<: *save_cache
+
+mdoc: &mdoc
+  steps:
+    - checkout
+    - <<: *load_cache
+    - run:
+        name: Generate documentation
+        command: ./sbt ++${SCALA_VERSION}! mdoc
+    - <<: *save_cache
+
+testJVM: &testJVM
+  steps:
+    - checkout
+    - <<: *load_cache
+    - <<: *install_jdk
+    - run:
+        name: Run tests
+        command: ./sbt ++${SCALA_VERSION}! testJVM
+    - run:
+        name: Package artifacts
```

Comment:
Could we do this only when we release? It takes 3-4 min for every test step.

Reply:
I don't see why not, just rip this `run` out. The release steps will package anyways if it has not happened already